### PR TITLE
Improve analytics data structure

### PR DESCRIPTION
**Pull request by yannickperrenet**
## Description

This PR introduces changes to our analytics module so that all data passes through the same logic, ensuring that we can easily work with the data thanks to the common structure. Additionally, the new implementation provides (inside the `_Anonymizer` class) an extensive overview of all the events that we support.

Structure always in the following format:
```
        {
            "event_properties": ...  # anonymized event properties
            "derived_properties": ...  # derived from removed properties
            "app_properties": ...  # Orchest application properties
        }
```

### How to test
1. `./orchest start --log-level=DEBUG`
2. Trigger some events
3. Check the `docker logs orchest-webserver` and note entries such as `Successfully sent analytics event 'session start'`.

In addition I checked the PostHog data to make sure entries are created.

### CODE REVIEW

File: services/orchest-webserver/app/app/analytics.py [hunk 1]
```diff
@@ -119,142 +340,20 @@ def send_anonymized_pipeline_definition(app, pipeline):
         step.pop("file_path")
         step_parameters_count += len(step.pop("parameters", {}))
 
-        env = step.get("environment", "")
-        if len(env):
+        env = step.get("environment")
+        if env is not None:
```
Comment (fruttasecca) [inline services/orchest-webserver/app/app/analytics.py:344]: Not sure if intended, but note that a step with no defined environment will have a step equal to `""`.

**Review by fruttasecca — APPROVED** [review]
LGTM, a nitpick: using an enum to formalize what events are supported would make it more clear and less error-prone given that you don't end up writing the actual event name when sending events, which would abstract away the need for bubbling up requirements like what an event name should look like. 

Much needed streamlining of the analytics module!

File: services/orchest-webserver/app/app/analytics.py (hunk 1 above)
Comment (yannickperrenet) [reply services/orchest-webserver/app/app/analytics.py]: Great catch. Fixed.

**Comment by yannickperrenet** [comment]
> using an enum to formalize what events are supported

I think this is great feedback. The top-level `send_event` function would become something like:
```python
analytics.send_event(
  app,
  analytics.EventName.JOB_CREATE,
 {},
)
```
Did I understand that correctly?

**Comment by fruttasecca** [comment]
>Did I understand that correctly?

Yes

About the exact pattern (`analytics.EventName.JOB_CREATE`), I have seen different flavours:
- `analytics.JOB_CREATE`
- `analytics.Event.JOB_CREATE`
- `analytics.EventName.JOB_CREATE`

I don't feel strongly about anything in particular, just enumerating.

**Comment by yannickperrenet** [comment]
The pre-commit `flake` hook fails with reason:

> F401 'os' imported but unused

But this is untrue seen that `os` is used [here](https://github.com/orchest/orchest/pull/394/files#diff-2954dbcbe7988e4447096015b9005b83085f14f7d4286ac33ffd0f4c8c92aec8R107).

**Comment by fruttasecca** [comment]
lgtm

**Pull request merged**